### PR TITLE
Argument inspector

### PR DESCRIPTION
**Pull request by marijanlekic**

### CODE REVIEW

File: src/app/tool-editor/sections/arguments/argument-inspector.component.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import {CommandArgumentModel} from "cwlts/models/d2sb";
+import {ComponentBase} from "../../../components/common/component-base";
+
+@Component({
+    selector: "ct-argument-inspector",
+    template: `
+<ct-form-panel [borderless]="true" [collapsed]="false">
+
+        <span class="tc-header">
+            Basic
```
Comment (mayacoda) [inline src/app/tool-editor/sections/arguments/argument-inspector.component.ts]: You can remove the `ct-form-panel` completely. It's the only section of the inspector, so it isn't necessary.

File: src/app/tool-editor/sections/arguments/argument-inspector.component.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        </span>
+        
+        <div class="tc-body">
+            <form [formGroup]="form">
+            
+                <!--Prefix Field-->
+                <div class="form-group">
+                    <label>Prefix:</label>
```
Comment (mayacoda) [inline src/app/tool-editor/sections/arguments/argument-inspector.component.ts]: Don't add `:` at the end of labels. Use the class `form-control-label` on them.

File: src/app/tool-editor/sections/arguments/argument-inspector.component.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                    [formControl]="form.controls['valueFrom']">                    
+                    </ct-expression-input>
+                </div>
+                
+                <!--Separator field-->            
+                <div class="form-group flex-container">
+                    <label>Separator:</label>
+                    <span class="align-right">
+                        {{form._value.separate? "Yes" : "No"}}
+                        <toggle-slider [formControl]="form.controls['separate']"></toggle-slider>
```
Comment (mayacoda) [inline src/app/tool-editor/sections/arguments/argument-inspector.component.ts]: `toggle-slider` has two new inputs, `[on]` and `[off]`, where you can pass in string for on and off state. Also, I changed the label and text for separator in the inputBinding, copy that one.

File: src/app/tool-editor/sections/arguments/argument-inspector.component.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                    </span>
+                </div>
+                
+                <!--Position Field-->
+                <div class="form-group">
+                    <label>Position:</label>
+                    <input type="text" type="number" class="form-control" [formControl]="form.controls['position']">
```
Comment (mayacoda) [inline src/app/tool-editor/sections/arguments/argument-inspector.component.ts]: Double attribute declaration. Should be number.

File: src/app/tool-editor/sections/arguments/argument-inspector.component.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    public save = new Subject<FormGroup>();
+
+    constructor(private formBuilder: FormBuilder) {
+        super();
+    }
+
+    ngOnInit() {
+        this.form = this.formBuilder.group({
+            valueFrom: new FormControl(this.argument.valueFrom),
+            separate: new FormControl(this.argument.separate || false),
```
Comment (mayacoda) [inline src/app/tool-editor/sections/arguments/argument-inspector.component.ts]: Default for separate is actually true (if it doesn't exist, it's taken to be true). Setting the value to `this.argument.separate !== false` should suffice.

File: src/app/tool-editor/sections/inputs/tool-input-list.component.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ import {
 import {ComponentBase} from "../../../components/common/component-base";
 import {CommandInputParameterModel} from "cwlts/models/d2sb";
 import {EditorInspectorService} from "../../../editor-common/inspector/editor-inspector.service";
-import {Subject} from "rxjs";
```
Comment (mayacoda) [inline src/app/tool-editor/sections/inputs/tool-input-list.component.ts]: Necessary import, should keep it :)

File: src/app/tool-editor/sections/arguments/argument-inspector.component.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import {Subject} from "rxjs";
+import {Component, Input, Output} from "@angular/core";
+import {FormBuilder, FormGroup, FormControl} from "@angular/forms";
+import {CommandArgumentModel} from "cwlts/models/d2sb";
+import {ComponentBase} from "../../../components/common/component-base";
+
+@Component({
+    selector: "ct-argument-inspector",
+    template: `        
+        <div class="tc-body">
```
Comment (mayacoda) [inline src/app/tool-editor/sections/arguments/argument-inspector.component.ts]: Also unnecessary, this div is part of the `form-panel` component.

**Comment by marijanlekic** [comment]
Input and Output inspectors are now in separated folders. Description, symbols and secondary files components are in common-sections folder.

**Pull request merged**